### PR TITLE
Replace update_ticket_step with additional_steps

### PR DESCRIPTION
**Pull request by hanstrompert**
The optional update_ticket_step argument of the create_workflow, modify_workflow and terminate_workflow decorators is implemented in a non-optional way. This is resolved by replacing it with an additional_steps argument that takes a StepList instead of a Step. Besides now being able to add more than one additional step, as will be used by the Generate Workflows CLI command, the step list can also be empty.

Note that this will be a breaking change for those who use update_ticket_step in their custom orchestrator.

### CODE REVIEW

File: orchestrator/workflows/utils.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def create_workflow(
     description: str,
     initial_input_form: Optional[InputStepFunc] = None,
     status: SubscriptionLifecycle = SubscriptionLifecycle.ACTIVE,
-    update_ticket_step: Optional[Step] = None,
+    additional_steps: StepList = begin,
```
Comment (Mark90) [inline orchestrator/workflows/utils.py:189]: Looks a bit odd for `begin` to be inserted in the middle of the workflow. Perhaps just default to `StepList()`?

**Review by Mark90 — APPROVED** [review]
We probably only use this ourselves for conditionally running a Jira Ticket update step if there's `ticket_id` in the state.
So in this case it's ok to skip the deprecation warning.

We do need to adapt our Surf code for this.

File: orchestrator/workflows/utils.py (hunk 1 above)
Comment (hanstrompert) [reply orchestrator/workflows/utils.py:189]: That is what I initially had 😃, but then ruff started to complain: `B008 Do not perform function call 'StepList' in argument defaults`, but I can off course ignore that warning.

Comment (Mark90) [reply orchestrator/workflows/utils.py:189]: Yeah, ruff is right, and I'm wrong. There shouldn't be a list in the argument defaults.

Better to default to `None` and evaluate in the function body with `additional_steps = additional_steps or StepList()`

**Comment by hanstrompert** [comment]
> We probably only use this ourselves for conditionally running a Jira Ticket update step if there's `ticket_id` in the state. So in this case it's ok to skip the deprecation warning.

It can also be used this to add an send_email() step for example.
 
> We do need to adapt our Surf code for this.

There already is a merge request open to adapt the SURF code.

File: orchestrator/workflows/utils.py (hunk 1 above)
Comment (hanstrompert) [reply orchestrator/workflows/utils.py]: Applied suggestion.

**Review by pboers1988 — APPROVED** [review]
LGTM, after conflict fix

**Comment by hanstrompert** [comment]
Conflicts were caused by premature merge of part of this pull request. Conflicts are solved.

**Pull request merged**